internal/subscriber/subscriber.go

```diff
@@ -207,7 +210,21 @@ func (s *Subscriber) acknowledge(req *AckMessage) {
 	}
 
 	if shouldCommit {
-		_, err := s.consumer.CommitByPartitionAndOffset(ctx, messagebroker.CommitOnTopicRequest{
+		offsetUpdated := true
+		registryOffset := strconv.Itoa(int(offsetToCommit) + 1)
+		offsetModel := offset.Model{
```

Comment:
don't we need to do the same thing in `subscriber.retry` when we are comitting on the primary topic?